File: src/main/java/com/github/jacekpoz/common/sendables/database/queries/user/DenyFriendRequestQuery.java
package com.github.jacekpoz.common.sendables.database.queries.user;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.jacekpoz.common.sendables.database.queries.basequeries.UserQuery;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class DenyFriendRequestQuery extends UserQuery {

    @Getter
    private final long friendID;

    @JsonCreator
    public DenyFriendRequestQuery(
            @JsonProperty("senderID") long senderID,
            @JsonProperty("friendID") long friendID,
            @JsonProperty("callerID") long callerID
    ) {
        super(senderID, callerID);
        this.friendID = friendID;
    }

}

File: src/main/java/com/github/jacekpoz/common/sendables/database/queries/user/DenyFriendRequestQuery.kt
package com.github.jacekpoz.common.sendables.database.queries.user

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.jacekpoz.common.sendables.database.queries.basequeries.UserQuery

open class DenyFriendRequestQuery @JsonCreator constructor(
    @JsonProperty("senderID") senderID: Long,
    @JsonProperty("friendID") private val friendID: Long,
    @JsonProperty("callerID") callerID: Long,
) : UserQuery(senderID, callerID) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is DenyFriendRequestQuery) return false
        if (!super.equals(other)) return false

        if (friendID != other.friendID) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + friendID.hashCode()
        return result
    }

    override fun toString(): String = "DenyFriendRequestQuery(senderID=$userID, friendID=$friendID, callerID=$callerID)"
}